modules/core/src/test/scala/org/tessellation/infrastructure/trust/storage/TrustStorageSuite.scala

```diff
@@ -0,0 +1,361 @@
+        )
+    }
+  }
+
+  test("internal trust storage is updated with predicted trusts") {
+    val gen = Gen.mapOfN(4, genPeerLabel)
+
+    forall(gen) { trusts =>
+      for {
+        store <- TrustStorage.make
+        firstTrust <- store.getTrust
+
+        _ <- store.updatePredictedTrust(trusts)
+
+        expected = trusts.view.mapValues(trustValue => TrustInfo().copy(predictedTrust = trustValue.some)).toMap
+        actual <- store.getTrust.map(_.trust)
+      } yield
+        expect.all(
+          TrustMap.empty == firstTrust,
+          expected == actual
+        )
+    }
+  }
+
+  test("internal trust storage is updated with public trust") {
+    val gen = for {
+      peerId <- peerIdGen
+      labels <- Gen.mapOfN(4, genPeerLabel)
+    } yield (peerId, PublicTrust(labels))
+
+    forall(gen) {
+      case (peerId, publicTrust) =>
+        for {
+          store <- TrustStorage.make
+          firstTrust <- store.getTrust
+
+          _ <- store.updatePeerPublicTrustInfo(peerId, publicTrust)
+
+          expected = PublicTrustMap(Map(peerId -> PublicTrust(publicTrust.labels)))
+          actual <- store.getTrust.map(_.peerLabels)
+        } yield
+          expect.all(
+            TrustMap.empty == firstTrust,
+            expected == actual
+          )
+    }
+  }
+
+  test("initial current ordinal trust storage is empty") {
+    for {
+      store <- TrustStorage.make
+      trust <- store.getCurrentOrdinalTrust
+    } yield expect.all(OrdinalTrustMap.empty == trust)
+  }
+
+  test("initial next ordinal trust storage is empty except for the ordinal") {
+    for {
+      store <- TrustStorage.make
+      trust <- store.getNextOrdinalTrust
+    } yield
+      expect.all(
+        TrustMap.empty == trust.trust,
+        SnapshotOrdinal(snapshotInterval) == trust.ordinal,
+        PublicTrustMap.empty == trust.peerLabels
+      )
+  }
+
+  test("next ordinal trust storage doesn't update from internal storage because expected ordinal has not been reached") {
+    val minOrdinalValue = 0L
+    val maxOrdinalValue = snapshotInterval - 1L
+
+    val gen = for {
+      trustInfo <- Gen.mapOfN(4, genPeerTrustInfo)
+      delta <- Gen.chooseNum(minOrdinalValue, maxOrdinalValue)
+      ordinal = Refined.unsafeApply[Long, NonNegative](delta)
+    } yield (TrustMap.empty.copy(trust = trustInfo), SnapshotOrdinal(ordinal))
+
+    forall(gen) {
+      case (trust, ordinal) =>
+        for {
+          store <- TrustStorage.make(trust)
+          storedTrust <- store.getTrust
+
+          maybeOrdinalPublicTrust <- store.updateNext(ordinal)
+
+          next <- store.getNextOrdinalTrust
+        } yield
+          expect.all(
+            trust == storedTrust,
+            SnapshotOrdinal(snapshotInterval) > ordinal,
+            maybeOrdinalPublicTrust.isEmpty,
+            TrustMap.empty == next.trust,
+            SnapshotOrdinal(snapshotInterval) == next.ordinal,
+            PublicTrustMap.empty == next.peerLabels
```

Comment:
aa

Reply:
All equality in the tests I created have been updated to use `===`.